### DISCUSSION TITLE
Importing Custom Shapes

### DISCUSSION THREAD

**Discussion by trayres** [hneemann/Digital#894] [Q&A] [answered]
I don't see how to import custom shapes. 

I export the SVG, open inkscape, make changes, save the changed file - but how do we actually trigger the import?

**Comment by xotmatrix** [comment, marked as answer]
Edit > Circuit specific settings > Advanced > Shape: User defined > Import

**Reply by trayres** [reply]
Thank you, that was driving me crazy.

**Reply by trayres** [reply]
I am having extreme difficulty getting items to line up correctly when imported to Digital. 

Does anyone have suggestions, hints, or "best practices" to make that work out? 

The display of strings between Inkscape and Digital is not consistent (it looks like things like subscripts are handled as plain strings in Inkscape, but are specially processed in Digital, leading to different dimensions within text objects).

**Reply by richardson1959** [reply]
I didn't realize that pins had to be placed at the intersection of major grid-lines in Inkscape.  It didn't take too long to figure that out though.  Inkscape has an alignment tool that you can use to line up different objects (strings).  Subscripts are a pain, but I've gotten better at guessing what the string will look like in Digital.

**Reply by hneemann** [reply]
Try to add a text like `Y_1^{n+1}` to the SVG file. 
I expect such a string to be formatted correctly if imported into the simulator.

**Reply by richardson1959** [reply]
Yes, it is formatted correctly.  I think the issue is that Inkscape does not support Digital syntax for subscripts/superscripts so the strings look very different in the two applications.

**Reply by trayres** [reply]
I was re-sizing the SVG document in Inkscape after making changes - this made it impossible to figure out where the pins "should be". I think it's implicitly reading from the upper left hand corner, which moves if you re-size the diagram.

NOT re-sizing the SVG, I was able to get the pins where I want them! Thanks everybody!

**Reply by LucianoSaldivia** [reply]
> I was re-sizing the SVG document in Inkscape after making changes - this made it impossible to figure out where the pins "should be". I think it's implicitly reading from the upper left hand corner, which moves if you re-size the diagram.
> 
> NOT re-sizing the SVG, I was able to get the pins where I want them! Thanks everybody!

I'm having trouble understanding what to do to make a working example of this. I just want to create a simple custom shape to start, and then modify similar ones. Let's say i want to make a 4to1 multiplexer with the trapezoid shape. How would i do this? 
I'd be happy if you send me any example file, or any useful tutorial or resource to look into, thanks in advance.
